### ISSUE TITLE
Fails to run with older versions of ESBuild

### ISSUE THREAD

**Issue by nayaabkhan** [contentlayerdev/contentlayer#205]
## Error

```
UnknownEsbuildError: Error: Build failed with 1 error:
[redacted]/node_modules/esbuild/lib/main.js:239:12: error: Invalid option in build() call: "jsx"
```

## Possible cause

**@contentlayer/core** lists an old version in `peerDependencies`:
https://github.com/contentlayerdev/contentlayer/blob/20ce9059ca7da34a2a7eb9b7650be2b6bcde6cdd/packages/%40contentlayer/core/package.json#L17

It uses the `jsx` option which was added in ESBuild [v0.12.1](https://github.com/evanw/esbuild/blob/master/CHANGELOG.md#0121):

https://github.com/contentlayerdev/contentlayer/blob/20ce9059ca7da34a2a7eb9b7650be2b6bcde6cdd/packages/%40contentlayer/core/src/getConfig/index.ts#L69

## Proposed fix

https://github.com/contentlayerdev/contentlayer/pull/204/files

**Comment by schickling** [comment]
Thanks for opening this issue. Can you please provide a reproduction repo with instructions how to reproduce the problem you're experiencing? Thank you!

**Comment by nayaabkhan** [comment]
Hey @schickling, here's the [reproduction repo](https://github.com/nayaabkhan/next-contentlayer-example) forked from the https://github.com/contentlayerdev/next-contentlayer-example.

There are two way this can be reproduced:
1. By adding another package that requires an older version of esbuild: https://github.com/nayaabkhan/next-contentlayer-example/commit/0471b02de5c37fd111caa04c4ff4e48c09659835
2. By using `overrides` to manually set a version: https://github.com/nayaabkhan/next-contentlayer-example/commit/fe9b5c4d5c729b99fa56f09057b5f8df45aa50ab

**Comment by schickling** [comment]
This should be addressed with the [`0.2.5` release](https://github.com/contentlayerdev/contentlayer/releases/tag/v0.2.5). 🎉
